src/client/mapper.ts

```diff
@@ -93,6 +102,55 @@ export function timeFilterSimpleToRequest(body: TimeFilterSimple): TimeFilterReq
   };
 }
 
+export function timeFilterSimpleToFullMatrix(body: TimeFilterSimple, max = 2000) {
+  const searchBase: Omit<TimeFilterRequestSearchBase, 'id'> = {
+    properties: body.properties,
+    transportation: body.transportation,
+    travel_time: body.travelTime,
+    range: body.range,
+  };
+
+  const searchType = body.searchType === 'arrive' ? 'arrival_searches' : 'departure_searches';
+  const origin = body.searchType === 'arrive' ? 'arrival_location_id' : 'departure_location_id';
+  const destinations = body.searchType === 'arrive' ? 'departure_location_ids' : 'arrival_location_ids';
+  const timeType = body.searchType === 'arrive' ? 'arrival_time' : 'departure_time';
+
+  if (body.locations.length > max) {
+    const searches: TimeFilterRequest[][] = [];
+    let i = 0;
+    while (i < body.locations.length) {
+      const search = body.locations.slice(i, i + max);
+      searches.push(search.map((loc, index) => ({
+        locations: search,
+        [searchType]: [{
+          ...searchBase,
+          id: `location-${index}`,
+          [origin]: loc.id,
+          [destinations]: [...search.slice(0, index).map((l) => l.id), ...search.slice(index + 1).map((l) => l.id)],
```

Comment:
I think you want smth like this?
```suggestion
[destinations]: search.toSpliced(i, 1)..map((l) => l.id)
```

Reply:
Will not use yet as its not LTS currently.